### PR TITLE
Add KeyboardActions parameter to KPInputPasswordField and KPInputOTPF…

### PR DESCRIPTION
**Pull request by alibardide5124**
### What does this PR do?
In `KPInputPasswordField` and `KPInputOTPField`, removed hard-coded keyboard action and take `KeyboardActions` as a parameter in these two functions, with a default value.

## 📋 Related Issue
Related to issue #18 

### How has this been tested?

- [ ] Screenshot tests
- [ ] UI tests
- [X] Manual testing
Run the application in a phone to check if desired behavior is achieved.

### Test Coverage
- [X] Main functionality
- [ ] Edge cases
- [ ] Different Android versions
- [ ] Different screen sizes/densities
- [ ] RTL layout support
- [ ] Accessibility features

### General
- [X] Code follows project coding standards
- [X] Self-review completed
- [X] No debugging code left behind
- [X] Documentation updated if needed

### Testing
- [X] Existing tests still pass
- [ ] New tests added if applicable
- [ ] Screenshot tests updated if UI changed
- [ ] Edge cases covered

### Breaking Changes
- [ ] This change introduces breaking changes
- [X] This change is backward compatible

## 🔄 Migration Required

- [X] No migration required
- [ ] Migration required (explain below)

**Migration Details:**

## 📸 Screenshot Test Changes

- [X] Screenshot tests were not affected
- [ ] Screenshot tests were updated
- [ ] New screenshot tests were added

**Screenshot Changes Explained:**

---

### CODE REVIEW

File: core/src/main/java/com/trendyol/design/core/inputfield/otp/InputOTPField.kt [hunk 1]
```diff
@@ -87,26 +90,19 @@ public fun KPInputOTPField(
...
-                .fillMaxWidth()
-                .onKeyEvent { keyEvent ->
-                    if (keyEvent.nativeKeyEvent.keyCode == KeyEvent.KEYCODE_ENTER) {
-                        focusManager.clearFocus(force = true)
-                        onValueChange(otpTextValue.text)
-
-                        return@onKeyEvent true
-                    }
-
-                    false
```
Comment (turkergoksu) [inline core/src/main/java/com/trendyol/design/core/inputfield/otp/InputOTPField.kt:100]: Do you remember why we added this onValueChange @mertkalecik 🤔

File: core/src/main/java/com/trendyol/design/core/inputfield/otp/InputOTPField.kt [hunk 2]
```diff
@@ -87,26 +90,19 @@ public fun KPInputOTPField(
...
+            modifier = Modifier.fillMaxWidth(),
             value = otpTextValue,
             onValueChange = { value ->
                 otpTextValue = value.copy(
                     text = value.text.trim(),
                     selection = TextRange(value.text.length)
                 )
             },
-            keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.NumberPassword),
+            keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.NumberPassword, imeAction = ImeAction.Done),
```
Comment (turkergoksu) [inline core/src/main/java/com/trendyol/design/core/inputfield/otp/InputOTPField.kt:101]: Nice addition 👏

File: core/src/main/java/com/trendyol/design/core/inputfield/otp/InputOTPField.kt [hunk 3]
```diff
@@ -87,26 +90,19 @@ public fun KPInputOTPField(
...
+            keyboardActions = keyboardActions ?: KeyboardActions(
+                onDone = { focusManager.clearFocus(force = true) },
+
```
Comment (turkergoksu) [inline core/src/main/java/com/trendyol/design/core/inputfield/otp/InputOTPField.kt:104]: We should delete the empty line

**Comment by alibardide5124** [comment]
> Also it would be better if you change the commit message to be in the format of issue number as pointed in [CONTRIBUTING.md](https://github.com/Trendyol/komposto/blob/main/CONTRIBUTING.md#%EF%B8%8F-commit-message-format)
> 
> `GH-18 | Add KeyboardActions parameter to KPInputPasswordField and KPInputOTPField`


Sorry I didn't noticed this part. All requested changes were made, though I'm not sure if it's supposed to automatically replace 'GH-18' with `TrendyolGH-18` at the beginning of the commit, or if it's an issue on my part.

**Comment by turkergoksu** [comment]
https://github.com/Trendyol/komposto/actions/runs/18339003792/job/52233374936?pr=19
I think you also need to update your branch with 
`./gradlew apiDump`

and commit the changed file which supposed to be `core.api` file. This is related to binary-compatibility. You may check out the https://github.com/Kotlin/binary-compatibility-validator if you want learn about its purpose.

Also a nitpick small suggestion for commit messages. Try to explain what you changed in that commit with a short message.

```
GH-18 | Move onValueChange call inside onDone function
GH-18 | Update core.api file
```

etc... This will help other developers to adapt your changes.

**Review by turkergoksu — APPROVED** [review]
Everything seems to be working. Thank you for your contribution. Approved ✅
